### PR TITLE
[Feature] Add script to clean up golden AMIs in AWS regions where we're at our quota for public images

### PR DESCRIPTION
**Pull request by AlexVulaj**
## What does this PR do? / Related Issues / Jira

Adds a Go script to clean up our Golden AMIs in regions where we've hit our quota.

## Checklist
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have tested the functionality against gcp / aws, it doesn't cause any regression
- [x] I have added execution results to the PR's readme

## How to test this PR locally / Special Instructions

First, make sure you have AWS credentials for the target account. Then cd into `cleangoldenami`, run `go build`, then run the executable (preferably with `--dry-run` and `--verbose`)

## Logs 

```
Region af-south-1 is under quota. (Images: 13, Quota: 20)
Region ap-east-1 is under quota. (Images: 14, Quota: 20)
Region ap-northeast-1 is under quota. (Images: 14, Quota: 20)
Region ap-northeast-2 is under quota. (Images: 13, Quota: 20)
Region ap-northeast-3 is under quota. (Images: 14, Quota: 20)
Region ap-south-1 is under quota. (Images: 14, Quota: 20)
Region ap-south-2 is under quota. (Images: 8, Quota: 20)
Region ap-southeast-1 is under quota. (Images: 15, Quota: 20)
Region ap-southeast-2 is under quota. (Images: 13, Quota: 20)
Region ap-southeast-3 is under quota. (Images: 13, Quota: 20)
Region ap-southeast-4 is under quota. (Images: 8, Quota: 20)
Region ca-central-1 is under quota. (Images: 13, Quota: 20)
Region eu-central-1 is at quota (10) - would delete ami-08eca5bab97189074
Region eu-central-2 is under quota. (Images: 8, Quota: 20)
Region eu-north-1 is under quota. (Images: 9, Quota: 20)
Region eu-south-1 is under quota. (Images: 12, Quota: 20)
Region eu-south-2 is under quota. (Images: 7, Quota: 20)
Region eu-west-1 is under quota. (Images: 13, Quota: 20)
Region eu-west-2 is under quota. (Images: 14, Quota: 20)
Region eu-west-3 is under quota. (Images: 14, Quota: 20)
Region me-central-1 is under quota. (Images: 6, Quota: 20)
Region me-south-1 is under quota. (Images: 12, Quota: 20)
Region sa-east-1 is under quota. (Images: 13, Quota: 20)
Region us-east-1 is under quota. (Images: 11, Quota: 20)
Region us-east-2 is under quota. (Images: 15, Quota: 20)
Region us-west-1 is under quota. (Images: 12, Quota: 20)
Region us-west-2 is under quota. (Images: 14, Quota: 20)
Done!
```

### CODE REVIEW

File: cleangoldenami/regions.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package main
+
+var regions = []string{
```
Comment (luis-falcon) [inline cleangoldenami/regions.go]: Can we have this whole list/file be replaced by an aws API call where we get all the regions?

File: cleangoldenami/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+that will loop through each region supported by the verifier
+and delete the oldest public AMI if the resource quota is at capacity.
+
+## Usage
+
+It can be run with AWS credentials setup as environment variables beforehand, e.g.
+
+```bash
+AMI_ACCOUNT_ID=...
+export $(osdctl account cli -i ${AMI_ACCOUNT_ID}-p rhcontrol -oenv | xargs)
```
Comment (luis-falcon) [inline cleangoldenami/README.md:13]: space between  ${AMI_ACCOUNT_ID} -p

File: cleangoldenami/regions.go (hunk 1 above)
Comment (AlexVulaj) [reply cleangoldenami/regions.go]: Great suggestion - done in latest commit (+ tests)

File: cleangoldenami/README.md (hunk 2 above)
Comment (AlexVulaj) [reply cleangoldenami/README.md]: Good catch! Fixed in latest commit.

**Comment by abyrne55** [comment]
Tested locally with `--dry-run` and saw expected results. Also ran `go test` and visually reviewed code. Nice work @AlexVulaj

/approve

**Pull request merged**